src/seedu/addressbook/Main.java

```diff
@@ -79,26 +88,62 @@ private void exit() {
     }
 
     /** Reads the user command and executes it, until the user issues the exit command.  */
-    private void runCommandLoopUntilExitCommand() {
+    private void runCommandLoopUntilExitCommand() throws UniquePersonList.DuplicatePersonException {
         Command command;
+        CommandResult result;
+        deletes = new PreviousDeletes();
         do {
             String userCommandText = ui.getUserCommand();
             command = new Parser().parseCommand(userCommandText);
-            CommandResult result = executeCommand(command);
+            result = getCommandResult(command, userCommandText);
             recordResult(result);
             ui.showResultToUser(result);
 
         } while (!ExitCommand.isExit(command));
     }
 
+    /**
+     *
```

Comment:
Better to get rid of these empty header comments as they don't add any new value.